### ISSUE TITLE
CMake branch?

### ISSUE THREAD

**Issue by discapes** [zelbrium/gllabel#16]
I made a branch that uses CMake (cross platform) and GLAD, a more modern alternative to GLEW. It has the includes in a separate folder, for the convenience of library users. It also follows the convention that users of the library do `lib/file`, (here `gllabel/label.hpp`, just like other libraries (`glm/vec2.hpp`, `GLFW/glfw3.h`, `glad/glad.h`)). The libary source code itself can refer to just `<label.hpp>` for the sake of less typing (that can be easily changed though). I also made simple build instructions in README.md, and a small script to run them all. I have also set the code up to bake the fonts right into the executable using a CMake script (CMakeRC), so the executable can be run everywhere. It really doesn't take much more than ~`FT_New_Memory_Face(cmrc::fonts.open("Sans-Regular.tff), ...)`. So I thought you might wanna make a CMake branch for users who prefer to use that, I could PR there...

**Comment by azsn** [comment]
So I'm not necessarily opposed to using GLAD, but i don't really see the point. GLEW fully supports up to the most modern version of OpenGL (my computer is actually running OpenGL 4.1, released in 2010, because Apple is garbage and doesn't support anything newer than that -_-). Also it looks like you're just using a default install of GLAD, but from what I can tell, the point of it is that it [lets you generate an extension loader](https://glad.dav1d.de/) that's specific to your needs. So if you're just using a default install I'm not really sure how it's any different from GLEW. Please let me know if I'm misunderstanding.

Putting the includes into a separate include directory is a good idea. Two things, though: First is that generally includes aren't inside `lib/`, they're next to it. So you'd have the root directory containing `include/` and `lib/` side-by-side. Second is that there's a difference between public and private headers. C++ makes the distinction kind of weird because `class` declarations can have private parts that must go in public headers, but generally you want header files that have internal usage to go in-line with the source files, while public headers go in the `include/` directory. For this project, I'm expecting that only `label.hpp` (probably should get renamed to `gllabel.hpp`) will be a public header and everything else is private. Sorry if I'm saying things you already know -- I'm not sure how experienced of a developer you are.

The embedded fonts thing is cool, but I'm expecting that anyone who uses this library would have a larger environment to work with and should embed fonts in the way that makes the most sense for them. This project isn't about the demo, its about the library, so I'd like to keep the demo part simple as possible -- just enough to show off the functionality of the library.

Similarly, I don't think CMake is worth the trouble for such a small library and demo program. Maybe instead just use `pkg-config`? It doesn't always work but it should get the job done on Mac and Linux, at least. (And it would probably be just as reliable as CMake for the needs of this project.) Also, just wondering, does your current `cmake` branch (3f1b466) actually build for you? You use `target_link_libraries` to load the dependencies, but I'm pretty sure that only works for libraries within the project. For locating external dependencies you need something like `find_libraries` (pretty similar to `pkg-config`), I think.

tl;dr I'm not going to pull in the changes on your `cmake` branch as-is. I'd accept a PR that puts headers in an `include/` directory, though, if the public/private thing I mentioned above is followed, and a separate PR that converts the makefile to use `pkg-config`, if you want to do that. Thanks!

**Comment by discapes** [comment]
As for GLAD vs GLEW, I see what you mean. I was thinking it might also be possible to decouple the library from either one, and maybe GLAD might be good for the demo app if I strip it down to include just the needed functions... oops mismisclicked the close, button, I still have stuff to write

**Comment by discapes** [comment]
Yeah I really ended up overcomplicating all the stuff with the headers. I agree that internal headers are just as .cpp files. I hadn't then looked at what headers were supposed to be private.

Well, I just gotta clean a lot of stuff, and yeah I can also probably make the prs u mentioned
